CDCgov/trusted-intermediary
# Use inject for non singleton class

# Description

This PR adds the logic necessary for non-singleton classes to use the @Inject annotation. 

# Changes
 `ApplicationContext.java`
  - Refactoring of `skipMissingImplementation` logic
  - Renamed `injectRegisteredImplementations` -> `doInjectRegisteredImplementations`
  - Overload `injectIntoField` method
  - Added `injectIntoNonSingleton()` method
  
 `Reflections.java`
  - Added `getFieldsAnnotatedWithInstance`
  
  `ApplicationContextText.groovy`
  - Added test cases
  
  `TestApplicationContext.java`
  - Added `skipMissingImplementation` logic
  
  `HapiFhirResource` 
  - The changes in this file were used for testing, they will get deleted once the PR is approved
  
  `HapiFhirResourceTest`
  - This file was used for testing, it will get deleted once the PR is approved

# Testing
- Choose a non-singleton class to work with. Make sure the class is not dynamically instantiated
- Use @Inject on fields that you want to inject the dependency
- Make sure the class sends itself as a parameter to`ApplicationContext.injectIntoNonSingleton()`, using the constructor
- Build module and make sure all tests pass

## Issue
#1361

## Checklist


**Note**: You may remove items that are not applicable
